spec/features/admin/applications/show_spec.rb

```diff
@@ -0,0 +1,169 @@
+require 'rails_helper'
+
+RSpec.describe "Admin Application Show Page" do
+  let!(:application_1) {Application.create!(name: "Sally", street_address: "112 W 9th St.", city: "Kansas City", state: "MO", zip_code: "64105", description: "I love animals. Please let me have one.", status: "pending")}
+  let!(:application_2) {Application.create!(name: "Marcus", street_address: "100 Hennepin Ave.", city: "Minneapolis", state: "MN", zip_code: "55401", description: "Dogs are the best. Please let me have one.", status: "pending")}
+
+  let!(:shelter_1) {Shelter.create!(foster_program: true, name: "Adopters Unite", city: "Minneapolis", rank: 1 ) }
+
+  let!(:pet_1) {shelter_1.pets.create!(adoptable: true, age: 3, breed: "doberman", name: "Rover")}
+  let!(:pet_2) {shelter_1.pets.create!(adoptable: true, age: 2, breed: "dalmatian", name: "Pongo")}
+
+  let!(:application_pet_1) {ApplicationPet.create!(pet: pet_1, application: application_1, pet_status: "pending")}
+  let!(:application_pet_2) {ApplicationPet.create!(pet: pet_2, application: application_1, pet_status: "pending")}
+  let!(:application_pet_3) {ApplicationPet.create!(pet: pet_1, application: application_2, pet_status: "pending")}
+  let!(:application_pet_4) {ApplicationPet.create!(pet: pet_2, application: application_2, pet_status: "pending")}
+
+  describe "As a visitor" do
+    describe "User Story 12" do
+      describe "When I visit an admin application show page" do
+        it "sees a button to approve each pet in the application, after I click on the button, I'm taken back to the admin application show page and see 'Approved' next to that pet" do
+          visit "/admin/applications/#{application_1.id}"
+
+          expect(page).to have_button("Approve", count: 2)
+          expect(page).to have_content(pet_1.name)
+          expect(page).to have_content(pet_2.name)
+
+          click_on "Approve", match: :first
```

Comment:
Great use of `match` for differentiating between buttons. This is also where you could make use of within blocks if you needed to approve only one